packages/nextjs/test/config.test.ts

```diff
@@ -324,6 +327,45 @@ describe('webpack config', () => {
       );
     });
   });
+
+  describe('`distDir` value in default server-side `RewriteFrames` integration', () => {
+    describe('`DefinePlugin` existence and inclusion of `__rewriteFramesDistDir__` definition', () => {
+      it.each([
+        ['no custom `distDir`, pre-existing `DefinePlugin`', undefined, true],
+        ['no custom `distDir`, no pre-existing `DefinePlugin`', undefined, false],
+        ['custom `distDir`, pre-existing `DefinePlugin`', 'some/output/directory', true],
+        ['custom `distDir`, no pre-existing `DefinePlugin`', 'some/output/directory', false],
+      ])(
+        '%s',
+        async (_testTitle: string, customDistDir: string | undefined, existingDefinePlugin: boolean): Promise<void> => {
+          const mightHaveDistDir = {
+            ...userNextConfig,
+            ...(customDistDir && { distDir: customDistDir }),
+          };
+          const mightHaveDefinePlugin = {
+            ...serverWebpackConfig,
+            ...(existingDefinePlugin && { plugins: [new DefinePlugin({ __dogName__: 'Maisey' })] }),
+          };
+
+          const finalWebpackConfig = await materializeFinalWebpackConfig({
+            userNextConfig: mightHaveDistDir,
+            incomingWebpackConfig: mightHaveDefinePlugin,
+            incomingWebpackBuildContext: getBuildContext('server', mightHaveDistDir),
+          });
+
+          expect(finalWebpackConfig.plugins).toEqual(expect.arrayContaining([expect.any(DefinePlugin)]));
+
+          const definePluginInstance = findWebpackPlugin(finalWebpackConfig, 'DefinePlugin') as DefinePlugin;
+          expect(definePluginInstance.definitions.__rewriteFramesDistDir__).toEqual(customDistDir);
```

Comment:
Why are we expecting to have a `DefinePlugin`, if we're later checking a property of it? The first one is redundant, and the check we really want to do is the latter; we can remove the first.

Reply:
I put in both so that if the latter fails, we know it's not because the plugin isn't there, it's because the definition is wrong. I can add a comment to that effect, though.

Reply:
I understand the reason for doing that. But, I think it's redundant and doesn't add value to the tests (it doesn't test additional branches, statements...). Again, not blocking the PR for this reason.